src/EvaluatorUtils.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2020, The Souffle Developers. All rights reserved.
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file EvaluatorUtils.h
+ *
+ * Defines utility functions used by synthesised and interpreter code.
```

Comment:
What's this file for? The description is a little vague.

I was planning to split up `Util.h` into `SpecificPurposeUtils.h` files, like `ContainerUtils.h` and `FileUtils.h`, so this fits right in.

Reply:
This is for common code that's used by RAM evaluators (e.g. the interpreter and synthesized code). This allows code reuse instead of having to quote helper functions in the synthesizer.

Reply:
We always talked about splitting the Utils into sub-files. That would be amazing starting this refactoring.

Reply:
Perhaps we are more strategic on how we split the functionality depending on data-structure, ast, ram, etc.